app/src/main/kotlin/com/github/gotify/api/CertUtils.kt

```diff
@@ -86,13 +110,23 @@ internal object CertUtils {
     }
 
     @Throws(GeneralSecurityException::class)
-    private fun newEmptyKeyStore(): KeyStore {
-        return try {
-            val keyStore = KeyStore.getInstance(KeyStore.getDefaultType())
-            keyStore.load(null, null)
-            keyStore
-        } catch (e: IOException) {
-            throw AssertionError(e)
-        }
+    private fun certToKeyManager(certPath: String, certPassword: String?): Array<KeyManager> {
+        require(certPassword != null) { "empty client certificate password" }
+
+        val keyStore = KeyStore.getInstance("PKCS12")
```

Comment:
I've debugged nearly an hour why this didn't worked for me and guess what the PKCS12 key provider didn't exist on my emulator. On my real device this works fine but on my emulator this error was logged. I couldn't really find anything online regarding to this, so it's probably some edge-case that doesn't happen in real devices.
<details><summary>Details</summary>
<p>

```
Failed to apply SSL settings: java.io.IOException: error constructing MAC: java.security.InvalidKeyException: No installed provider supports this key: com.android.org.bouncycastle.jcajce.PKCS12Key
    at com.android.org.bouncycastle.jcajce.provider.keystore.pkcs12.PKCS12KeyStoreSpi.engineLoad(PKCS12KeyStoreSpi.java:852)
    at java.security.KeyStore.load(KeyStore.java:1484)
    at com.github.gotify.api.CertUtils.certToKeyManager(CertUtils.kt:118)
    at com.github.gotify.api.CertUtils.applySslSettings(CertUtils.kt:63)
    at com.github.gotify.api.ClientFactory.defaultClient(ClientFactory.kt:80)
    at com.github.gotify.api.ClientFactory.unauthorized(ClientFactory.kt:21)
    at com.github.gotify.api.ClientFactory.versionApi(ClientFactory.kt:49)
    at com.github.gotify.login.LoginActivity.doCheckUrl(LoginActivity.kt:147)
    at com.github.gotify.login.LoginActivity.onPostCreate$lambda$6(LoginActivity.kt:118)
    at com.github.gotify.login.LoginActivity.$r8$lambda$YPO6IepZ2o2Scolja7nou-1QsOA(Unknown Source:0)
    at com.github.gotify.login.LoginActivity$$ExternalSyntheticLambda4.onClick(D8$$SyntheticClass:0)
    at android.view.View.performClick(View.java:7455)
    at com.google.android.material.button.MaterialButton.performClick(MaterialButton.java:1218)
    at android.view.View.performClickInternal(View.java:7432)
    at android.view.View.access$3700(View.java:835)
    at android.view.View$PerformClick.run(View.java:28810)
    at android.os.Handler.handleCallback(Handler.java:938)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:201)
    at android.os.Looper.loop(Looper.java:288)
    at android.app.ActivityThread.main(ActivityThread.java:7870)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

</p>
</details>

Reply:
Oh no, I've actually never tried it with an emulator :(   
I suppose the MAC generation (via [bouncy castle](https://www.bouncycastle.org/java.html)) is supported by the hardware security chip which does not exists in an emulator.

Reply:
We could disallow mTLS on emulators.

Reply:
What do you think? Should be block mTLS functionality on emulators?

Reply:
I'm fine with keeping it, but disabling the cert selection when we detect an emulator is also fine.

Reply:
Then let's keep it for now